spring-integration-file/src/main/java/org/springframework/integration/file/remote/synchronizer/AbstractInboundFileSynchronizingMessageSource.java

```diff
@@ -145,6 +151,18 @@ public void setUseWatchService(boolean useWatchService) {
 		}
 	}
 
+	/**
+	 * Switch the local {@link FileReadingMessageSource} to use a custom
+	 * {@link DirectoryScanner}.
+	 * @param scanner the {@link DirectoryScanner} to use.
+	 * @since 5.0
+	 */
+	public void setScanner(DirectoryScanner scanner) {
+		Assert.notNull(scanner, "scanner must not be null");
+		this.fileSource.setScanner(scanner);
```

Comment:
The `this.fileSource.setScanner(scanner);` takes care about assert. 
We don't need this extra.